libyul/AsmJsonImporter.cpp

```diff
@@ -52,9 +52,11 @@ template <class T>
 T AsmJsonImporter::createAsmNode(Json::Value const& _node)
 {
 	T r;
-	r.location = createSourceLocation(_node);
+	r.debugData = createDebugData(createSourceLocation(_node));
 	yulAssert(
-		r.location.source && 0 <= r.location.start && r.location.start <= r.location.end,
+		r.debugData->location.source &&
```

Comment:
I think we may be better off with something like the following, because we are trying to ensure the `createSourceLocation` helper made up good values. Or potentially we could just move this assertion into `createDebugData` (assuming we use it everywhere).
```
auto location = createSouceLocation(_node);
assert(location....);

T.r;
r.debugData = createDebugData(location);
return ret;
```

Reply:
Refactored, should be looking much better